src/main/java/org/tikv/service/failsafe/CircuitBreakerImpl.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2021 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.tikv.service.failsafe;
+
+import io.prometheus.client.Counter;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.tikv.common.TiConfiguration;
+
+public class CircuitBreakerImpl implements CircuitBreaker {
+  private static final Logger logger = LoggerFactory.getLogger(CircuitBreakerImpl.class);
+
+  private static final Counter CIRCUIT_BREAKER_ATTEMPT_COUNTER =
+      Counter.build()
+          .name("client_java_circuit_breaker_attempt_counter")
+          .help("client circuit breaker attempt counter.")
+          .labelNames("type")
+          .register();
+
+  private final boolean enable;
+  private final int windowInSeconds;
+  private final int errorThresholdPercentage;
```

Comment:
why not use double to store the error rate threshold, since it would be more concise?

Reply:
In this case, it's enough to use an integer to represent Percentage.

Reply:
`Hystrix` use `errorThresholdPercentage` as integer type [0, 100].